numpy/core/src/multiarray/buffer.c

```diff
@@ -483,7 +483,25 @@ _buffer_info_new(PyObject *obj)
         goto fail;
     }
 
-    if (PyArray_IsScalar(obj, Generic)) {
+    if (PyArray_IsScalar(obj, Datetime) || PyArray_IsScalar(obj, Timedelta)) {
+        info->ndim = 1;
```

Comment:
I'd prefer this appear after the comment, alongside all the other assigments

Reply:
fixed